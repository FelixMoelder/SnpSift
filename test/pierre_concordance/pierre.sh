#!/bin/sh

java -jar SnpSift.jar concordance myref.vcf myvcf.vcf > res.txt

awk 'BEGIN{cv["0/0"]=0;cv["0|0"]=0;cv["0/1"]=1;cv["1/0"]=1;cv["0|1"]=1;cv["1|0"]=1;cv["1/1"]=2;cv["1|1"]=2; \
    cv["."]="M";cv["./."]="M";cv[".|."]="M";cv["0|."]="M";cv["1|."]="M";cv[".|0"]="M";cv[".|1"]="M";cnt=0;} \
    FNR==NR{if($0!~/^#/){id=$1":"$2;split($10,a,":");x1[id]=cv[a[1]]; if(id in y){pp=0;} else {cnt++;y[id]=cnt;z[cnt]=id;}} next} \
    {if($0!~/^#/){id=$1":"$2;split($10,a,":");x2[id]=cv[a[1]]; if(id in y){pp=0;} else {cnt++;y[id]=cnt;z[cnt]=id;}}} \
    END{cnt2=0;for(i=1;i<=cnt;i++){b1=b2="A";idd=z[i];if(idd in x1){b1=x1[idd];}if(idd in x2){b2=x2[idd];} \
    gg=b1"_"b2; if(gg in f){f[gg]++;} else {f[gg]=1;cnt2++;h[cnt2]=gg;}} \
    for(i=1;i<=cnt2;i++){print(h[i], f[h[i]]);}} ' \
    myref.vcf myvcf.vcf
