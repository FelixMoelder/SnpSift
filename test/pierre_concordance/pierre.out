

SnpSift
chr	#
pos	Total
ref	0
alt	21
MISSING_ENTRY_myref/MISSING_ENTRY_myvcf	764
MISSING_ENTRY_myref/MISSING_GT_myvcf	3
MISSING_ENTRY_myref/REF	0
MISSING_ENTRY_myref/ALT_1	0
MISSING_ENTRY_myref/ALT_2	0
MISSING_GT_myref/MISSING_ENTRY_myvcf	0
MISSING_GT_myref/MISSING_GT_myvcf	0
MISSING_GT_myref/REF	0
MISSING_GT_myref/ALT_1	0
MISSING_GT_myref/ALT_2	0
REF/MISSING_ENTRY_myvcf	0
REF/MISSING_GT_myvcf	0
REF/REF	0
REF/ALT_1	1
REF/ALT_2	0
ALT_1/MISSING_ENTRY_myvcf	0
ALT_1/MISSING_GT_myvcf	8
ALT_1/REF	0
ALT_1/ALT_1	0
ALT_1/ALT_2	1
ALT_2/MISSING_ENTRY_myvcf	0
ALT_2/MISSING_GT_myvcf	0
ALT_2/REF	3
ALT_2/ALT_1	
ALT_2/ALT_2	


Pierre
1_1 8
2_2 4
2_M 1
2_A 118
1_A 260
_A 20
A_0 764
A_M 21
A_1 2
A_ 1
